portal/src/views/shared/Comments.vue

```diff
@@ -0,0 +1,371 @@
+                                Reply
+                            </button>
+                            <button v-if="viewType === 'data'" @click="viewDataClick(post)">
+                                <img src="@/assets/icon-view-data.svg">
+                                View Data
+                            </button>
+                        </div>
+                    </div>
+                </div>
+            </transition-group>
+        </div>
+    </section>
+</template>
+
+<script lang="ts">
+
+import Vue from "vue";
+import CommonComponents from "@/views/shared";
+import moment from "moment";
+import {Bookmark} from "@/views/viz/viz";
+
+export interface Comment {
+    id: number,
+    author: object,
+    body: string;
+    replies: [];
+    createdAt: number;
+    updatedAt: number;
+}
+
+export default Vue.extend({
+    name: "Comments",
+    components: {
+        ...CommonComponents,
+    },
+    props: {
+        user: {
+            required: true,
+        },
+        parentData: {
+            required: true,
+            type: [Number, Bookmark],
+        },
+    },
+    methods: {
+        getNewCommentPlaceholder(): string {
+            if (this.viewType === "project") {
+                return "Comment on Project";
+            } else {
+                return "Write a comment about this Data View";
+            }
+        },
+        save(comment): void {
+            this.$services.api
+                .postComment(comment)
+                .then((response: {post: Comment}) => {
+                    this.newComment.body = null;
+                    // add the comment to the replies array
+                    if (comment.threadId) {
+                        this.posts.filter((post) => post.id === comment.threadId)[0].replies.push(response.post);
+                        this.newReply.body = null;
+                    } else {// add it to the posts array
+                        this.posts.unshift(response.post);
+                        this.newComment.body = null;
+                    }
+                }).catch(e => {
+                    this.errorPostComment = true;
+                });
+        },
+        formatTimestamp(timestamp): string {
+            return moment(timestamp).fromNow();
+        },
+        addReply(post): void {
+            this.newReply.threadId = post.id;
+            this.newReply.body = null;
+        },
+        getComments(): void {
+            this.$services.api
+                .getComments(this.parentData)
+                .then((data) => {
+                    this.posts = data.posts;
+                })
+                .catch(e => {
+                    this.errorGetComments = true;
+                });
+        },
+        viewDataClick(post) {
+            this.$emit("viewDataClicked", JSON.parse(post.bookmark));
+        },
+    },
+    mounted(): void {
+        this.placeholder = this.getNewCommentPlaceholder();
+        this.getComments();
+    },
+    data(): any {
```

Comment:
I'm avoiding the use of 'any' because it just leads to false confidence. So if you could strongly type this return, as well too please. Thanks!

Reply:
I'm missing the point of specifying a return type for data(). Is there any instance in which the returned value from this function could be used?